src/java/org/apache/cassandra/schema/CompressionParams.java

```diff
@@ -97,128 +109,185 @@
-    {
-        return new CompressionParams(DeflateCompressor.instance, chunkLength, Integer.MAX_VALUE, 0.0, Collections.emptyMap());
-    }
+        String sstableCompressionClass = removeSstableCompressionClass(options);
 
-    @VisibleForTesting
-    public static CompressionParams lz4()
-    {
-        return lz4(DEFAULT_CHUNK_LENGTH);
+        return fromClassAndOptions(sstableCompressionClass, options);
     }
 
-    @VisibleForTesting
-    public static CompressionParams lz4(int chunkLength)
+    private static CompressionParams fromClassAndOptions(String sstableCompressionClass, Map<String,String> options)
     {
-        return lz4(chunkLength, chunkLength);
-    }
+        boolean enabled = true;
 
-    @VisibleForTesting
-    public static CompressionParams lz4(int chunkLength, int maxCompressedLength)
-    {
-        return new CompressionParams(LZ4Compressor.create(Collections.emptyMap()), chunkLength, maxCompressedLength, calcMinCompressRatio(chunkLength, maxCompressedLength), Collections.emptyMap());
-    }
+        if (!removeEnabled(options))
+        {
+            enabled = false;
+        }
 
-    public static CompressionParams zstd()
-    {
-        return zstd(DEFAULT_CHUNK_LENGTH);
+        int chunk_length_in_kb = removeChunkLength(options);
+
+        // figure out how we calculate the max_compressed_length and min_compress_ratio
+        if (options.containsKey(MIN_COMPRESS_RATIO)  && options.containsKey(MAX_COMPRESSED_LENGTH))
+        {
+            throw new ConfigurationException("Can not specify both 'min_compress_ratio' and 'max_compressed_length' for the compressor parameters.");
+        }
+
+        // calculate the max_compressed_length and min_compress_ratio
+        int max_compressed_length;
+        double min_compress_ratio;
+        String max_compressed_length_str = options.remove( MAX_COMPRESSED_LENGTH);
+        if (!StringUtils.isBlank(max_compressed_length_str))
+        {
+            try
+            {
+                max_compressed_length = new DataStorageSpec.IntKibibytesBound(max_compressed_length_str).toKibibytes();
+                validateMaxCompressedLength( max_compressed_length, chunk_length_in_kb);
+                min_compress_ratio = CompressionParams.calcMinCompressRatio(chunk_length_in_kb, max_compressed_length);
+            } catch (IllegalArgumentException e) {
+                throw new ConfigurationException(invalidValue( MAX_COMPRESSED_LENGTH, e.getMessage(), max_compressed_length_str ));
+            }
+        }
+        else
+        {
+            min_compress_ratio = removeMinCompressRatio(options);
+            validateMinCompressRatio( min_compress_ratio );
+            max_compressed_length =  CompressionParams.calcMaxCompressedLength(chunk_length_in_kb,min_compress_ratio);
+        }
+
+        // try to set compressor type
+        CompressorType compressorType = StringUtils.isEmpty(sstableCompressionClass)?CompressorType.lz4:null;
```

Comment:
`CompressorType compressorType = StringUtils.isEmpty(sstableCompressionClass) ? CompressorType.lz4 : null;`

Here some spaces would be handy!